src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,426 @@
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreatePromtMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_.ManageHMICommand(tts_global_properties);
+  }
+}
+
+void HelpPromptManagerImpl::SendUIRequest() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  LOG4CXX_DEBUG(logger_, "UI request for appID:" << app_.app_id());
+  smart_objects::SmartObjectSPtr ui_global_properties =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  if (ui_global_properties) {
+    smart_objects::SmartObject& ref = *ui_global_properties;
+
+    ref[strings::params][strings::message_type] =
+        hmi_apis::messageType::request;
+    ref[strings::params][strings::protocol_version] =
+        commands::CommandImpl::protocol_version_;
+    ref[strings::params][strings::protocol_type] =
+        commands::CommandImpl::hmi_protocol_type_;
+    ref[strings::params][strings::correlation_id] =
+        app_manager_.GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *ui_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        hmi_apis::FunctionID::UI_SetGlobalProperties;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreateVRMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_.ManageHMICommand(ui_global_properties);
+  }
+}
+
+void HelpPromptManagerImpl::SendBothRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  SendTTSRequest();
+  SendUIRequest();
+}
+
+void HelpPromptManagerImpl::SendRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+
+  switch (sending_type_) {
+    case SendingType::kSendHelpPromt:
+      SendTTSRequest();
+      break;
+    case SendingType::kSendVRHelp:
+      SendUIRequest();
+      break;
+    case SendingType::kSendBoth:
+      SendBothRequests();
+      break;
+    default:
+      LOG4CXX_DEBUG(logger_,
+                    "SendingType:" << (unsigned)sending_type_
+                                   << " request not sending");
+      return;
+  }
+}
+
+void HelpPromptManagerImpl::CreatePromtMsg(
+    smart_objects::SmartObject& out_msg_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  out_msg_params[strings::help_prompt] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  VRCommandsMap::const_iterator it = vr_commands_.begin();
+  uint32_t index = 0;
+  for (; it != vr_commands_.end(); ++it) {
```

Comment:
maybe better to use 
for (const auto& it : vr_commands_) {

Reply:
Thanks, fixed in https://github.com/smartdevicelink/sdl_core/pull/2186/commits/1de0eb183562592396687a028c121176c53c0aeb